src/System.Net.Http/tests/FunctionalTests/HttpRequestMessageTest.cs

```diff
@@ -215,6 +217,43 @@ public void ToString_DefaultAndNonDefaultInstance_DumpAllFields()
                 "}", rm.ToString());
         }
 
+        [Theory]
+        [InlineData("DELETE")]
+        [InlineData("OPTIONS")]
+        [InlineData("HEAD")]
+        public async Task HttpRequest_BodylessMethod_NoContentLength(string method)
+        {
+            if (IsWinHttpHandler || IsNetfxHandler || IsUapHandler)
+            {
+                // Some platform handlers differ but we don't take it as failure.
+                return;
+            }
+
+            using (HttpClient client = new HttpClient())
+            {
+                await LoopbackServer.CreateServerAsync(async (server, uri) =>
+                {
+                    var request = new HttpRequestMessage();
+                    request.RequestUri = uri;
+                    request.Method = new HttpMethod(method);
+
+                    Task<HttpResponseMessage> requestTask = client.SendAsync(request);
+                    await server.AcceptConnectionAsync(async connection =>
+                    {
+                        List<string> headers = await connection.ReadRequestHeaderAsync();
+                        foreach (string line in headers)
+                        {
+                            // There should be no Content-Length header.
+                            Assert.False(line.StartsWith("Content-length:", StringComparison.InvariantCultureIgnoreCase));
```

Comment:
You should use `Assert.DoesNotContain` pattern instead of `Assert.False`.

See example:
https://github.com/dotnet/corefx/blob/4b3564c2b3142ade08325689a8ff1476f4616b8f/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs#L953

Reply:
Why is that better?

Reply:
It produces better output when things fail.  And it is pattern we already use elsewhere in the tests.

Reply:
ok. Existing pattern also exit but I can change it as you suggested.